src/protocols/ssh/input.c

```diff
@@ -31,6 +33,20 @@
 
 #include <pthread.h>
 
+#include <stdio.h>
+#include <sys/timeb.h>
+
+
+int state = 0;
+long long start_1, start_2, start_3 = 0;
+long long end_1, end_2, end_3 = 0;
+
+long long getSystemTime() {
+    struct timeb t;
+    ftime(&t);
+    return 1000 * t.time + t.millitm;
+}
```

Comment:
You do not need to declare your own function for this - libguac provides `guac_timestamp_current()`.